### ISSUE TITLE
makefile: easier overrides when invoking make

### ISSUE THREAD

**Issue by arathkone** [soimort/translate-shell#277]
**TL;DR**: use `?=` instead of `=` in Makefile so `FOO=bar make stuff` works properly.

Using `=` in Makefile variable assignments like BUILDDIR and PREFIX prevents environment overrides on the commandline, as in the pattern `FOO=bar make stuff`, which is a common recommendation.  One could use `make FOO=bar stuff`, which is a different pattern and may not be known to the user.  Or `FOO=bar make -e stuff`, which is less commonly known.  Switching to `?=` will allow overrides from the environment for variables that should be easily accessible.  I would suggest this switch in the definitions of BUILDDIR, PREFIX, and TARGET, if not the NAME and COMMAND as well.

A simple testcase:

```make
# add to Makefile to check override status
env:
        @echo "checking: BUILDDIR=$(BUILDDIR)"
        @echo "checking: PREFIX=$(PREFIX)"
```

```sh
# test commands: all should show overridden values

# this one doesn't work with current Makefile
$ BUILDDIR=/tmp/build PREFIX=/tmp make env

# incorrect output
checking: BUILDDIR=build
checking: PREFIX=/usr/local

# these both work
$ make BUILDDIR=/tmp/build PREFIX=/tmp env
$ BUILDDIR=/tmp/build PREFIX=/tmp make -e env

# correct output
checking: BUILDDIR=/tmp/build
checking: PREFIX=/tmp
```

Patch:
```patch
diff --git a/Makefile b/Makefile
index bc82afc..cb25f2a 100644
--- a/Makefile
+++ b/Makefile
@@ -3 +3 @@ COMMAND  = trans
-BUILDDIR = build
+BUILDDIR ?= build
@@ -6,2 +6,2 @@ MANDIR   = man
-TARGET   = bash
-PREFIX   = /usr/local
+TARGET   ?= bash
+PREFIX   ?= /usr/local
```

(edit: Removing MANDIR from this recommendation.  It is only used to locate the sourcefile, not specify the installation location.)

**Comment by soimort** [comment]
Thanks for the suggestion!
